src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo)
+                .append(System.lineSeparator());
+        for (String name : names) {
+            int userSalary = 0;
+            for (String row : data) {
+                String[] dataSplit = row.split(" ");
+                String foundDate = dataSplit[0];
+                String foundName = dataSplit[1];
+                String foundHours = dataSplit[2];
+                String foundHourSalary = dataSplit[3];
+                if (name.equals(foundName) && isDateIncluded(dateFrom, dateTo, foundDate)) {
+                    userSalary += Integer.parseInt(foundHours) * Integer.parseInt(foundHourSalary);
```

Comment:
Consider handling potential `NumberFormatException` when parsing integers from `foundHours` and `foundHourSalary`. This will make the code more robust against malformed data.